src/integrationtests/java/com/aws/greengrass/integrationtests/deviceauth/EvaluateClientDeviceActionsWithPolicyVariablesTest.java

```diff
@@ -0,0 +1,187 @@
+        kernel = new Kernel();
+
+        // Set up Iot auth client
+        IotAuthClientFake iotAuthClientFake = new IotAuthClientFake();
+        kernel.getContext().put(IotAuthClient.class, iotAuthClientFake);
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        startNucleusWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startNucleusWithConfig(String configFileName, State expectedServiceState) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        GlobalStateChangeListener listener = (GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME)
+                    && service.getState().equals(expectedServiceState)) {
+                authServiceRunning.countDown();
+            }
+        };
+        kernel.getContext().addGlobalStateChangeListener(listener);
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+        kernel.getContext().removeGlobalStateChangeListener(listener);
+        startCDAWithConfig();
+    }
+
+    private void startCDAWithConfig() {
+        // use DeviceAuthClient with mocked sessionManager
+        GroupManager groupManager = kernel.getContext().get(GroupManager.class);
+        CertificateStore certificateStore = kernel.getContext().get(CertificateStore.class);
+        deviceAuthClient = new DeviceAuthClient(sessionManager, groupManager, certificateStore);
+    }
+
+    private Session createSession(String thingName) throws Exception {
+        Certificate cert = CertificateFake.of("FAKE_CERT_ID");
+        Thing thing = Thing.of(thingName);
+        return new SessionImpl(cert, thing);
+    }
+
+    private AuthorizationRequest buildAuthZRequest(Session session, String operation, String topic) {
+        AuthorizeClientDeviceActionRequest connectRequest =
+                new AuthorizeClientDeviceActionRequest().withOperation(operation).withResource(String.format(topic, session.getSessionAttribute("Thing", "ThingName")));
+
+        return AuthorizationRequest.builder().sessionId("sessionId")
+                .operation(connectRequest.getOperation()).resource(connectRequest.getResource()).build();
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_cdaWithDeviceGroupPolicy_WHEN_AuthorizeClientDeviceValidResourceAction_THEN_returnDecision() throws Exception {
+
+        // CDA configuration with device group policies
+        startNucleusWithConfig("policyVariableCda.yaml");
+
+       when(sessionManager.findSession("sessionId")).thenReturn(createSession("myThing"));
```

Comment:
General comment, these still feel like unit tests--do we need all of this mocking of sessions, etc? Ideally we can just start CDA (using fakes for web service calls) with a config yml file and then confirm the state with an action a user would perform, like making an IPC call, and verify the request was authorized or not (think along the lines of testing inputs/outputs of CDA as a whole rather than of a single class).